core/src/main/java/org/axonframework/config/DefaultEventProcessorRegistry.java

```diff
@@ -0,0 +1,263 @@
+
+package org.axonframework.config;
+
+import org.axonframework.common.Assert;
+import org.axonframework.common.transaction.NoTransactionManager;
+import org.axonframework.common.transaction.TransactionManager;
+import org.axonframework.eventhandling.DirectEventProcessingStrategy;
+import org.axonframework.eventhandling.ErrorHandler;
+import org.axonframework.eventhandling.EventHandlerInvoker;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.eventhandling.EventProcessor;
+import org.axonframework.eventhandling.MultiEventHandlerInvoker;
+import org.axonframework.eventhandling.PropagatingErrorHandler;
+import org.axonframework.eventhandling.SubscribingEventProcessor;
+import org.axonframework.eventhandling.TrackedEventMessage;
+import org.axonframework.eventhandling.TrackingEventProcessor;
+import org.axonframework.eventhandling.TrackingEventProcessorConfiguration;
+import org.axonframework.eventhandling.tokenstore.TokenStore;
+import org.axonframework.eventhandling.tokenstore.inmemory.InMemoryTokenStore;
+import org.axonframework.messaging.Message;
+import org.axonframework.messaging.MessageHandlerInterceptor;
+import org.axonframework.messaging.StreamableMessageSource;
+import org.axonframework.messaging.SubscribableMessageSource;
+import org.axonframework.messaging.unitofwork.RollbackConfigurationType;
+import org.axonframework.monitoring.MessageMonitor;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+
+/**
+ * Implementation of {@link EventProcessorRegistry}.
+ *
+ * @author Milan Savic
+ * @since 3.3
+ */
+public class DefaultEventProcessorRegistry implements EventProcessorRegistry {
+
+    private final Map<String, List<Function<Configuration, EventHandlerInvoker>>> invokerBuilders = new HashMap<>();
+    private final Map<String, EventProcessorBuilder> eventProcessorBuilders = new HashMap<>();
+    private final Map<String, Component<EventProcessor>> eventProcessors = new HashMap<>();
+    private final Map<String, List<Function<Configuration, MessageHandlerInterceptor<? super EventMessage<?>>>>> handlerInterceptorsBuilders = new HashMap<>();
+    private final Map<String, Function<Configuration, ErrorHandler>> errorHandlers = new HashMap<>();
+    private final Map<String, MessageMonitorFactory> messageMonitorFactories = new HashMap<>();
+    private final Map<String, Function<Configuration, TokenStore>> tokenStore = new HashMap<>();
+    private EventProcessorBuilder defaultEventProcessorBuilder = this::defaultEventProcessor;
+    private Configuration configuration;
+
+    private final Component<ErrorHandler> defaultErrorHandler = new Component<>(
+            () -> configuration,
+            "errorHandler",
+            c -> c.getComponent(ErrorHandler.class, PropagatingErrorHandler::instance)
+    );
+
```

Comment:
Should this guy also be in charge of the `defaultListenerInvocationErrorHandler` currently in the `EventHandlingConfiguration`?

Reply:
Thus that mean we lose the ability to set a default `ListenerInvocationErrorHandler` for our `EventHandlerInvokers`? Or am I missing something from this PR where that functionality has been moved to.

Reply:
My idea was to leave instantiation of `SimpleEventHandlerInvoker` in `EventHandlingConfiguration`. And to register those invokers with `EventProcessorRegistry` during initialization. So, `ListenerInvocationErrorHandler` would stay in `EventHandlingConfiguration` since its purpose would be to take care of configuration of event handlers.

Reply:
Ah, I was missing that the `EventHandlingConfiguration` as a whole isn't deprecated, just the majority of what it did.